### ISSUE TITLE
"Field is not searchable" when getting record with EA

### ISSUE THREAD

**Issue by sjberman** [infobloxopen/infoblox-go-client#48]
I'm attempting to get a record (either A or CNAME, using GetObject), by specifying an object like this:

```
	obj := ibclient.NewRecordA(
		ibclient.RecordA{
			Name:     "foo.com",
			Ipv4Addr: "1.2.3.4",
			View:     "default",
			Ea:      ibclient.EA{eaKey: eaVal},
		},
	)
```
The body of the HTTP request that gets constructed looks like this;
```
'{"ipv4addr":"1.2.3.4","name":"foo.com","view":"default","extattrs":{"My-EA":{"value":"testing"}}}'
```

However, I'm getting this error: 
```
{ "Error": "AdmConProtoError: Field is not searchable: extattrs",
  "code": "Client.Ibap.Proto",
  "text": "Field is not searchable: extattrs"
}
```
The extattrs are not searchable. I need them to be, as all of the objects I create will have "My-EA", and at a certain point I'll have a general GetObject call to get all objects that I've created, and nothing more. Is this possible?

**Comment by chinmayb** [comment]
You'll have to use 'eaSearch' while creating RecordA object

**Comment by sjberman** [comment]
Can you provide an example? I'm not sure what you mean.

**Comment by sjberman** [comment]
If you're referring to setting the obj.eaSearch to something, the field is not exported, so I don't have access to setting the variable.

**Comment by chinmayb** [comment]
I'll raise a PR soon

**Comment by sjberman** [comment]
Thanks.

I did try to adjust the URL to have the EAKey and EAvalue in it (similar to how you mentioned), but it returned every object with the EA, and seemed to ignore the http body which had the details about the specific object I wanted. In this case I then had to loop through all those objects until I got the one that I cared about.

**Comment by chinmayb** [comment]
PR Raised https://github.com/infobloxopen/infoblox-go-client/pull/49
Can you check whether that works for you ?

**Comment by sjberman** [comment]
In briefly glancing at the code, I see a few things.

1. I need to be able to get both A and CNAME records.
2. The method you added, GetRecordA, accepts an EA and returns all A records. Which means I'll have to parse through and find the one I want (unless I want all of them, which I do in some cases). One of the nice things about the current GetObject method is that I can specify an A or CNAME record with specific fields, and I'll get back just that record. Being able to simply add an EA to the object I've constructed, and calling GetObject on that should return just that object (or a list of just that object, as it currently works).

Ideally I would like to be able to construct an A/CNAME record like the one above, and call GetObject and just get that object, or I could also construct an object like this:
```
obj := ibclient.NewRecordA(
		ibclient.RecordA{
			View:     "default",
			Ea:      ibclient.EA{eaKey: eaVal},
		},
	)
```
and get all A records in the "default" view with my EA, and still use GetObject on that. This is the current way of getting objects without specifying an EA.

**Comment by chinmayb** [comment]
> The method you added, GetRecordA, accepts an EA and returns all A records.

It returns all A records with the associated EA tag. You just have to pass it to the GetRecordA method.

**Comment by sjberman** [comment]
Right, I meant it returns all A records with the EA. What I would like is to simply call GetObject with an A/CNAME record that has an EA in its spec (along with other fields), and just get that single object back. That way I don't need to parse through all the returned objects to try and find the one that I wanted. Also, if I wanted to, I could create some generic A record like the one in my last comment, still use GetObject on it, and that would return me multiple A records. That is how I am currently using GetObject, it simply returns any objects that match the constructed object I passed in.

**Comment by yuewko** [comment]
@chinmayb , have you tried implementing GetRecordA() in a similar fashion as GetNetwork()? i.e. allow GetRecordA() to pass in commonly used search parameters such as netview, name, ipaddr etc, in addition to EA?

**Comment by sjberman** [comment]
Ideally what I would prefer is to continue using GetObject. It is convenient in that I can construct any record (A or CNAME), with whatever fields I want, and it will return to me any objects (or a single object) that match my passed in record. The enhancement that I'm asking for is to also be able to provide an EA 
 field in the record that I pass to GetObject.

**Comment by yuewko** [comment]
@sjberman Yes, that would be the way to do it. I haven't been working on this code for a while, so @chinmayb please see what the implication is.

So, basically, what we can do is, instead of doing, for example, the following in the GetNetwork() method:
```
    if ea != nil && len(ea) > 0 {                                                               
        network.eaSearch = EASearch(ea)                                                         
    }
```
move that into the GetObject() method.

**Comment by chinmayb** [comment]
@sjberman @yuewko should be doable. Will provide a fix as soon as I get some spare time.

**Comment by chinmayb** [comment]
For some objects EA field wouldnt be there. I feel why not use SetEaSearch field (which i included in the latest PR) explicitly to set the EAs ?

**Comment by yuewko** [comment]
Based on the comment https://github.com/infobloxopen/infoblox-go-client/issues/48#issuecomment-374647578, the expectation is to be able to construct an object with the desired search fields, and pass that to GetObject() to perform search accordingly.

If EA field is set, the GetObject() method should recognize it and set `eaSearch` field accordingly.
